tests/test_cli_compile.py

```diff
@@ -2702,9 +2706,225 @@ def test_all_extras_fail_with_extra(fake_dists, runner, make_module, fname, cont
     assert exp in out.stderr
 
 
+def _mock_resolver_cls(monkeypatch: MonkeyPatch) -> MagicMock:
+    obj = MagicMock()
+    obj.resolve = MagicMock(return_value=set())
+    obj.resolve_hashes = MagicMock(return_value=dict())
+    cls = MagicMock(return_value=obj)
+
+    monkeypatch.setattr("piptools.scripts.compile.BacktrackingResolver", cls)
+    monkeypatch.setattr("piptools.scripts.compile.LegacyResolver", cls)
+
+    return cls
+
+
+def _mock_build_project_metadata(monkeypatch: MonkeyPatch) -> MagicMock:
+    func = MagicMock(
+        return_value=ProjectMetadata(
+            extras=("e",),
+            requirements=(
+                install_req_from_line("rdep0"),
+                install_req_from_line("rdep1; extra=='e'"),
+            ),
+            build_requirements=(install_req_from_line("bdep0"),),
+        )
+    )
+
+    monkeypatch.setattr("piptools.scripts.compile.build_project_metadata", func)
+
+    return func
+
+
+@backtracking_resolver_only
```

Comment:
Could you please elaborate on why build-dist tests are only for the backtracking resolver?

Reply:
The functionality being tested does not depend on the resolver so testing it for both is redundant. This seemed like a worthwhile optimization for the slow `test_all_extras_and_all_build_deps` (previously `test_build_distribution`) but maybe applying it to the faster tests exposes us to unnecessary risk. What do you think?

Reply:
Thanks for the info. I agree with you, let's keep it as is. I really doubt that we need to add new tests for the legacy resolver.